python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -1016,3 +1036,52 @@ def _evaluate_auto_materialize_tick(
             )
 
         self._logger.info(f"Finished auto-materialization tick{print_group_name}")
+
+
+def invoke_sensor_for_evaluation(
+    sensor: ExternalSensor,
+    workspace_process_context: IWorkspaceProcessContext,
+    instigator_data: SensorInstigatorData,
+    evaluation_id: int,
+    stored_cursor: AssetDaemonCursor,
+    tick: InstigatorTick,
+    asset_graph: RemoteAssetGraph,
+) -> Tuple[Sequence[RunRequest], AssetDaemonCursor, Sequence[AssetConditionEvaluation]]:
+    sensor_origin = sensor.get_external_origin()
+    request_ctx = workspace_process_context.create_request_context()
+    code_loc = request_ctx.get_code_location(sensor_origin.location_name)
+
+    sensor_cursor = AssetDaemonCursor(
+        evaluation_id=evaluation_id,  # bump this before sending it over
+        previous_evaluation_state=stored_cursor.previous_evaluation_state,
+        last_observe_request_timestamp_by_asset_key=stored_cursor.last_observe_request_timestamp_by_asset_key,
+    )
+    sensor_cursor_str = asset_daemon_cursor_to_instigator_serialized_cursor(sensor_cursor)
```

Comment:
potentially a dragon here with a giant serialized string in certain situations right?